tests/python/relay/test_type_infer.py

```diff
@@ -193,7 +193,16 @@ def test_equal():
     # This should fail ....
     func = relay.Function([i], eq, ret_type=relay.TensorType([], 'int32'))
 
+ 
+def test_dup_type():
+    a = relay.TypeVar("a")
+    av = relay.Var("av", a)
+    make_dup = relay.Function([av], relay.Tuple([av, av]), None, [a])
+    t = relay.TensorType((10,))
+    b = relay.Var("b", t)
+    assert relay.ir_pass.infer_type(make_dup(b)).checked_type == relay.TupleType([t, t])
```

Comment:
I believe `==` isn't overridden for Types (only exprs) so you need to explicitly call `alpha_equal` here

Reply:
it is the opposite - it is overriden for type, not for expr.

Reply:
Lol